### PR TITLE
Authenticator "modal" variation

### LINKED ISSUE #576: Document Authenticator "modal" variation
- [x] #509 
- [x] #516
- [x] Under `Configuration`, document `variation` with the `modal` option.
- [x] What is the non-modal variation called? `default` doesn't make sense, since `modal` _is_ our default? (Or should it not be? And only `withAuthenticator` does `modal` by default?)
- [x] Update React to specify `data-authenticator-variation`
- [x] Update Angular to specify `data-authenticator-variation`
- [x] Update Vue to specify `data-authenticator-variation`

### PR DESCRIPTION
**Pull request by ericclemmons**
*Issue #, if available:* Closes #576

*Description of changes:*

- [x] Document `variation="modal"`

	> Screen Shot 2021-11-17 at 1 55 19 PM
	> Screen Shot 2021-11-17 at 1 55 13 PM

- [x] Add `variation` to Angular

	> Screen Shot 2021-11-17 at 1 10 00 PM
	> Screen Shot 2021-11-17 at 1 09 10 PM

- [x] Add `variation` to React

	> Screen Shot 2021-11-17 at 1 12 18 PM
	> Screen Shot 2021-11-17 at 1 13 06 PM

- [x] Add `variation` to Vue

	> Screen Shot 2021-11-17 at 1 10 26 PM
	> Screen Shot 2021-11-17 at 1 09 18 PM

- [x] Remove unnecessary example styles that made default variation look like a modal
- [x] Rename `withAuthenticator` test to `modal`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export class AuthenticatorComponent implements OnInit, AfterContentInit {
   @Input() services: AuthenticatorMachineOptions['services'];
   @Input() signUpAttributes: AuthenticatorMachineOptions['signUpAttributes'];
   @Input() socialProviders: SocialProvider[];
-  @Input() variation: 'modal' | undefined;
+  @Input() variation: 'default' | 'modal';
```
Comment (ErikCH) [inline packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts:32]: This is great, `default` and `modal` make sense

**Review by ErikCH — APPROVED** [review]
Tested, on the Vue side. Looks good. React, and Angular too, but will let @wlee221 double check that.

File: packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts (hunk 1 above)
Comment (ericclemmons) [reply packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts:32]: In the future, we may even go so far as `inline`, `ghost`, `plain`, or something if the card styles are too problematic to remove via CSS.

File: packages/angular/projects/ui-angular/src/lib/components/authenticator/components/sign-in/sign-in.component.html [hunk 2]
```diff
@@ -28,11 +28,14 @@
...
-      amplify-button
-      fontWeight="normal"
-      size="small"
-      variation="link"
       (click)="authenticator.toResetPassword()"
+      class="amplify-button amplify-field-group__control"
+      data-fullwidth="true"
+      data-size="small"
+      data-variation="link"
+      data-amplify-button=""
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/components/authenticator/components/sign-in/sign-in.component.html:36]: Quick question, why did you need to use `data-*`? `<button amplify-button fullWidth="true" />` to my knowledge.

https://github.com/aws-amplify/amplify-ui/blob/6dd6bf915b83e623dc9e526be7805dee15b9af8f/packages/angular/projects/ui-angular/src/lib/primitives/button/button.component.ts#L8-L12

File: packages/e2e/features/ui/components/authenticator/modal.feature [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
-  @react
+  @skip @angular @react @vue
   Scenario: Application is wrapped with Authenticator
     Then I see "Sign in"
 
-  @react
+  @skip @angular @react @vue
   Scenario: Application renders when signed in
     When I type my "username" with status "CONFIRMED"
     And I type my password
```
Comment (wlee221) [inline packages/e2e/features/ui/components/authenticator/modal.feature:15]: In the future, I think we can enhance this test to test for *actual* model behavior.

File: docs/src/pages/components/authenticator/quick-start.react.mdx [hunk 4]
```diff
@@ -1,6 +1,23 @@
...
+  <TabItem title="Authenticator">
+
+```js{3-4,11-12} file=../../../../../examples/next/pages/ui/components/authenticator/sign-in-with-username/index.page.tsx
+
+```
+
+  </TabItem>
+  <TabItem title="withAuthenticator">
+
+The `withAuthenticator` automatically sets the [`variation`](#variation) to `modal`.
```
Comment (wlee221) [inline docs/src/pages/components/authenticator/quick-start.react.mdx:16]: I was going to ask you if you should document what `withAuthenticator` uses for its variation, but you're already ahead of me 😄

**Review by wlee221 — APPROVED** [review]
LGTM! Thanks for the Angular example too.

File: packages/angular/projects/ui-angular/src/lib/components/authenticator/components/sign-in/sign-in.component.html (hunk 2 above)
Comment (ericclemmons) [reply packages/angular/projects/ui-angular/src/lib/components/authenticator/components/sign-in/sign-in.component.html:36]: Honestly, this was a copy paste from the Vue implementation.

I completely missed that this was an `amplify-button` directive!

I'll attempt this again, BRB...

File: packages/e2e/features/ui/components/authenticator/modal.feature (hunk 3 above)
Comment (ericclemmons) [reply packages/e2e/features/ui/components/authenticator/modal.feature:15]: In React, there's some default text in the app that we could do:

```
And I don't see "Example App"
```

https://github.com/aws-amplify/amplify-ui/blob/038075d4e47a5e213f44ab7798821b9c483be131/examples/next/pages/_document.page.tsx

Would that work?

File: packages/angular/projects/ui-angular/src/lib/components/authenticator/components/sign-in/sign-in.component.html (hunk 2 above)
Comment (ericclemmons) [reply packages/angular/projects/ui-angular/src/lib/components/authenticator/components/sign-in/sign-in.component.html]: That works!  Sorry @wlee221 !

Fixed via cb2577ea31f88d42bface6a8b2f55c38110b8856

File: docs/src/pages/components/authenticator/quick-start.react.mdx (hunk 4 above)
Comment (ericclemmons) [reply docs/src/pages/components/authenticator/quick-start.react.mdx:16]: There's an open issue (#640) to prefer `Authenticator` over `withAuthenticator`, so this was the 1st step towards that goal 💪

File: packages/e2e/features/ui/components/authenticator/modal.feature (hunk 3 above)
Comment (ericclemmons) [reply packages/e2e/features/ui/components/authenticator/modal.feature:15]: Actually, I'd rather have percy.io & visual regression tests cover this.

https://docs.cypress.io/guides/core-concepts/interacting-with-elements#Visibility isn't accurate enough to detect _it's not actually visible_.
